### PR TITLE
[Doc] Add `label` prop for `<ReferenceInput>` and `<ReferenceArrayInput>` in docs

### PR DESCRIPTION
**Pull request by thibault-barrat**
- [x] Fix doc mentionning that ReferenceInput does not need the "label" prop

### CODE REVIEW

File: docs/ReferenceArrayInput.md [hunk 1]
```diff
@@ -46,13 +46,13 @@ import { ReferenceArrayInput, SelectArrayInput } from 'react-admin';
 | Prop               | Required | Type                                        | Default                            | Description                                                                                                         |
 |--------------------|----------|---------------------------------------------|------------------------------------|---------------------------------------------------------------------------------------------------------------------|
 | `filter`           | Optional | `Object`                                    | `{}`                               | Permanent filters to use for getting the suggestion list                                                            |
+| `label`            | Optional | `string`                                    | -                                  | A custom label can be used when `ReferenceArrayInput` is in a Filter array.                         |
```
Comment (slax57) [inline docs/ReferenceArrayInput.md:49]: I would rather formulate this by saying this is useful **only** when `ReferenceArrayInput` is in a Filter array

File: docs/ReferenceInput.md [hunk 2]
```diff
@@ -26,13 +26,13 @@ import { ReferenceInput, SelectInput } from 'react-admin';
 | Prop               | Required | Type                                        | Default                          | Description                                                                                                       |
 |--------------------|----------|---------------------------------------------|----------------------------------|-------------------------------------------------------------------------------------------------------------------|
 | `filter`           | Optional | `Object`                                    | `{}`                             | Permanent filters to use for getting the suggestion list                                                          |
+| `label`            | Optional | `string`                                    | -                                | A custom label can be used when `ReferenceInput` is in a Filter array.                         |
```
Comment (slax57) [inline docs/ReferenceInput.md:29]: same

File: docs/Upgrade.md [hunk 3]
```diff
@@ -2709,6 +2709,8 @@ const UserInput = [
 ]
 ```
 
+**Note**: `<ReferenceInput>` and `<ReferenceArrayInput>` still accepts `label` prop only when they are used in a Filter array.
```
Comment (slax57) [inline docs/Upgrade.md:2712]:
```suggestion
**Note**: `<ReferenceInput>` and `<ReferenceArrayInput>` still accept the `label` prop, which is used as the Filter label when they are used in a Filter array.
```

**Review by slax57 — CHANGES_REQUESTED** [review]

**Review by septentrion-730n — APPROVED** [review]

**Review by slax57 — APPROVED** [review]
Niiice!

**Pull request merged**